encoding/cbf/common_codec/common.go

```diff
@@ -0,0 +1,505 @@
+	err = EncodeLocationPrefix(w, common.AddressLocationPrefix)
+	if err != nil {
+		return
+	}
+
+	err = EncodeAddress(w, t.Address)
+	if err != nil {
+		return
+	}
+
+	return EncodeString(w, t.Name)
+}
+
+func EncodeIdentifierLocation(w io.Writer, t common.IdentifierLocation) (err error) {
+	err = EncodeLocationPrefix(w, common.IdentifierLocationPrefix)
+	if err != nil {
+		return
+	}
+
+	return EncodeString(w, string(t))
+}
+
+func EncodeScriptLocation(w io.Writer, t common.ScriptLocation) (err error) {
+	err = EncodeLocationPrefix(w, common.ScriptLocationPrefix)
+	if err != nil {
+		return
+	}
+
+	_, err = w.Write(t[:])
+	return
+}
+
+func EncodeStringLocation(w io.Writer, t common.StringLocation) (err error) {
+	err = EncodeLocationPrefix(w, common.StringLocationPrefix)
+	if err != nil {
+		return
+	}
+
+	return EncodeString(w, string(t))
+}
+
+func EncodeTransactionLocation(w io.Writer, t common.TransactionLocation) (err error) {
+	err = EncodeLocationPrefix(w, common.TransactionLocationPrefix)
+	if err != nil {
+		return
+	}
+
+	_, err = w.Write(t[:])
+	return
+}
+
+func EncodeREPLLocation(w io.Writer) (err error) {
+	return EncodeLocationPrefix(w, common.REPLLocationPrefix)
+}
+
+func DecodeLocation(r io.Reader, memoryGauge common.MemoryGauge) (location common.Location, err error) {
+	prefix, err := DecodeLocationPrefix(r)
+
+	switch prefix {
+	case common.AddressLocationPrefix:
+		return DecodeAddressLocation(r, memoryGauge)
+	case common.IdentifierLocationPrefix:
+		return DecodeIdentifierLocation(r, memoryGauge)
+	case common.ScriptLocationPrefix:
+		return DecodeScriptLocation(r, memoryGauge)
+	case common.StringLocationPrefix:
+		return DecodeStringLocation(r, memoryGauge)
+	case common.TransactionLocationPrefix:
+		return DecodeTransactionLocation(r, memoryGauge)
+	case string(common.REPLLocationPrefix[0]):
+		location = common.REPLLocation{}
+	case NilLocationPrefix:
+		return
+
+	// TODO more locations
+	default:
+		err = CodecError(fmt.Sprintf("unknown location prefix: %s", prefix))
+	}
+	return
+}
+
+func DecodeLocationPrefix(r io.Reader) (prefix string, err error) {
+	b := make([]byte, 1)
+	_, err = r.Read(b)
+	prefix = string(b)
+	return
+}
+
+func DecodeAddressLocation(r io.Reader, memoryGauge common.MemoryGauge) (location common.AddressLocation, err error) {
+	address, err := DecodeAddress(r)
+	if err != nil {
+		return
+	}
+
+	name, err := DecodeString(r)
+	if err != nil {
+		return
+	}
+
+	location = common.NewAddressLocation(memoryGauge, address, name)
+
+	return
+}
+
+func DecodeIdentifierLocation(r io.Reader, memoryGauge common.MemoryGauge) (location common.IdentifierLocation, err error) {
+	s, err := DecodeString(r)
+	location = common.NewIdentifierLocation(memoryGauge, s)
+	return
+}
+
+func DecodeScriptLocation(r io.Reader, memoryGauge common.MemoryGauge) (location common.ScriptLocation, err error) {
+	byteArray := make([]byte, len(location)) // len(common.ScriptLocation) == 32
```

Comment:
might be worth making constants for location lengths, in case they become structs down the road that len() doesn't support. you're taking the length of a zero value here which seems brittle